Classes/Event/EventNormalizer.php

```diff
@@ -0,0 +1,57 @@
+<?php
+namespace Neos\EventSourcing\Event;
+
+/*
+ * This file is part of the Neos.EventSourcing package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Annotations as Flow;
+use Symfony\Component\Serializer\Normalizer\DateTimeNormalizer;
+use Symfony\Component\Serializer\Normalizer\JsonSerializableNormalizer;
+use Symfony\Component\Serializer\Normalizer\ObjectNormalizer;
+use Symfony\Component\Serializer\Serializer;
+
+/**
+ * @Flow\Scope("singleton")
+ *
+ * Converts instances of EventInterface to serializable arrays vice versa
+ */
+final class EventNormalizer
+{
+
+    /**
+     * @Flow\Inject
+     * @var EventTypeResolver
+     */
+    protected $eventTypeResolver;
+
+    /**
+     * @var Serializer
+     */
+    private $serializer;
+
+    protected function initializeObject(): void
+    {
+        // TODO: make normalizers configurable
+        $normalizers = [new DateTimeNormalizer(\DateTime::ISO8601), new JsonSerializableNormalizer(), new ObjectNormalizer()];
+        $this->serializer = new Serializer($normalizers);
+    }
+
+    public function normalize(EventInterface $event): array
+    {
+        return $this->serializer->normalize($event);
+    }
+
+    public function denormalize(array $eventData, string $eventType): EventInterface
+    {
+        // TODO allow to hook into event type => class conversion in order to enable upcasting, ...
+        $eventClassName = $this->eventTypeResolver->getEventClassNameByType($eventType);
```

Comment:
So the question is, if we should make the whole `EventNormalizer`/`EventConverter` configurable, or just the serializer and the eventTypeResolver separately.

In any case, I'd suggest to aim for a bijective API, i.e. `$normalizer->denormalize($normalizer->normalize($event)/*, get_class($event)*/) === $event`. This currently isn't the case, because the denormalization does an additional step (resolve the type input to an actual class), that needs to be done on the outside in the other direction. So additional information is needed about the internal workings of this normalizer to make use of it, which is not good.

So either we move the type resolving out of the normalizer again, or we make it return a normalized form that contains the resolved event type too (a serialized version of `WriteableEvent`?)